code/game/objects/items/storage/boxes.dm

```diff
@@ -438,6 +438,80 @@
 	for(var/i in 1 to 6)
 		new /obj/item/reagent_containers/food/snacks/donkpocket(src)
 
+/obj/item/storage/box/donkpockets/donkpocketspicy
+	name = "box of spicy-flavoured donk-pockets"
+	icon_state = "donkpocketboxspicy"
+	
+/obj/item/storage/box/donkpockets/donkpocketspicy/ComponentInitialize()
+	. = ..()
+
+/obj/item/storage/box/donkpockets/donkpocketspicy/PopulateContents()
+	for(var/i in 1 to 6)
+		new /obj/item/reagent_containers/food/snacks/donkpocket/spicy(src)

```

Comment:
Instead of copypasting this for every box just make the path variable here.

Reply:
Should be all good now!